### ISSUE TITLE
EditLink is wrong in 5.7

### ISSUE THREAD

**Issue by wkit23** [OData/WebApi#543]
In order to obtain the URL of a referred entity, referred to by a navigation property we use the full metadata like e.g.

```
http://localhost/MyResource(1)/MyNavigationProperty?$format=application/json;odata.metadata=full
```

which will return the following result in 5.6

```
{
    "@odata.editLink":"http://localhost/MyNavigationProperty(1)"
}
```

However when we upgraded to 5.7, the returned result became

```
{
    "@odata.editLink":"http://localhost/MyResource(1)/MyNavigationProperty(1)"
}
```

which apparently is wrong. 

I suspect this is caused by [https://github.com/OData/WebApi/pull/372]. Any idea? We mainly use this link to patch the odata entity.

**Comment by VikingsFan** [comment]
@ificator I've found that your PR #372 is for navigation prop, not only for contained navigation prop, so it's a breaking change for `id` and `editlink`, I'm going to fix this bug but I'm not sure it will break your service or not because you just add one test case. Can your tell me your scenario or provide a fix?

**Comment by VikingsFan** [comment]
@wkit23 Would you like to review my PR #554 ? Thanks!

**Comment by wkit23** [comment]
@VikingsFan I tried using your PR #554. It seems to work perfectly. Thanks!

**Comment by VikingsFan** [comment]
Merged 86e222607410a7bb61698b8f2ce69a0b1f24d901
